### PR TITLE
Improvements (refactor, memory, tests)

### PR DESCRIPTION
**Pull request by fkm**
A collection of improvements.

* Fist some minor refactoring where I replaced `foo: function () {` with `foo() {`.
* [Then the important part with the fixes/improvements for the memory leaks.](https://github.com/geoadmin/web-mapviewer/commit/d87bab631231cf13456d5ec65b109855966cc896)
* And a fix for the geolocation test that failed because of `watchPosition`.
* And the removal of an `.only` that I (again...) forgot to remove.

[Test link](https://web-mapviewer.dev.bgdi.ch/improvements/index.html)

### CODE REVIEW

File: src/modules/map/components/openlayers/utils/addLayerToMap-mixins.js [hunk 1]
```diff
@@ -26,8 +26,13 @@ const addLayerToMapMixin = {
     },
     unmounted() {
         if (this.layer && this.isPresentOnMap) {
+            this.layer.getSource()?.clear?.()
```
Comment (fkm) [inline src/modules/map/components/openlayers/utils/addLayerToMap-mixins.js:29]: I'm a bit on the fence about this notation.
It's simple and short. But (at least for now) unfamiliar. Especially the `clear?.()` part.

Feel free to tell me to do it the classic way! :smiley:

File: src/modules/map/components/openlayers/OpenLayersMap.vue [hunk 2]
```diff
@@ -237,19 +237,46 @@ export default {
...
+        this.map.un('pointerdown', this.onMapPointerDown)
+        this.map.un('pointerup', this.onMapPointerUp)
+        this.map.un('singleclick', this.onMapSingleClick)
+        this.map.un('pointerdrag', this.onMapPointerDrag)
+        this.map.un('moveend', this.onMapMoveEnd)
+        this.map.setTarget(null)
+        this.map.setView(null)
+
+        delete this.map
+        delete this.view
```
Comment (fkm) [inline src/modules/map/components/openlayers/OpenLayersMap.vue:260]: I'm not sure if this is necessary. I'd say that in theory those properties should go away automatically when the component is destroyed.

File: src/modules/map/components/openlayers/utils/addLayerToMap-mixins.js (hunk 1 above)
Comment (pakb) [reply src/modules/map/components/openlayers/utils/addLayerToMap-mixins.js:29]: I'm not sure you need the last `?.()` part, `this.layer.getSource()?.clear()` should be good enough (`clear()` will only be called if `getSource()` gives something)

**Review by pakb — APPROVED** [review]

File: src/modules/map/components/openlayers/utils/addLayerToMap-mixins.js (hunk 1 above)
Comment (fkm) [reply src/modules/map/components/openlayers/utils/addLayerToMap-mixins.js:29]: The problem is that not all sources have a clear method. So it is possible that `getSource()` returns something but `.clear` is undefined.

I might be thinking too far. But from what I see we have the following options:
* Optional chaining  
  Short but unfamiliar.
* Try/catch with an empty catch  
  Somewhat hacky.
* If-block  
  More lines but easy to read.

**Comment by fkm** [comment]
I allowed myself to amend the relevant commit. Here are the changes:

A compromise between an if-block and optional chaining.
```js
// src/modules/map/components/openlayers/utils/addLayerToMap-mixins.js
    unmounted() {
        if (this.layer && this.isPresentOnMap) {
            if (this.layer.getSource()?.clear) {
                this.layer.getSource().clear()
            }
            this.layer.setSource(null)

            this.removeLayerFromMap(this.layer)
        }

        delete this.layer
    },
```

And I moved the whole teardown to `unmounted`.
```js
// src/modules/map/components/openlayers/OpenLayersMap.vue
    unmounted() {
        this.map.un('pointerdown', this.onMapPointerDown)
        this.map.un('pointerup', this.onMapPointerUp)
        this.map.un('singleclick', this.onMapSingleClick)
        this.map.un('pointerdrag', this.onMapPointerDrag)
        this.map.un('moveend', this.onMapMoveEnd)
        this.map.setTarget(null)
        this.map.setView(null)

        delete this.map
        delete this.view
    },
```

**Pull request merged**